documentation/high-level-introduction.md

```diff
@@ -0,0 +1,257 @@
+
+Instead of including every single detail of a build in the user projects, this information tends to be split into multiple places. Some of these are provided by SDKs are stored in standard global locations, for example: `/Library/Frameworks/Mono.framework/External/xbuild/Xamarin/` 
+
+
+- Project files (csproj)
+  - These are user provided files describing the files, references, and settings for the build in question. 
+  - These tend to contain many Reference items and additions to the Compile and None item group
+  - These tend to include one single project specific “target” file that include the rest of the files needed to define the entire build.
+
+Examples:
+
+`<Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />`
+`<Import Project="$(MSBuildExtensionsPath)\Xamarin\Mac\Xamarin.Mac.CSharp.targets" />`
+
+- Solution files (.sln) combine one or more projects for editing and building. They contain some build dependency and configuration information, but project files are the main level of build knowledge.
+- Target Files (.targets)
+  - These files are generally provided by the SDKs
+  - These files generally describe a set of steps (targets) which need to be executed to process the build
+  - These tend to contain many UsingTask items to reference task assembly targets and Target to define the targets and dependencies
+- Property file (.props)
+  - These files are generally provided by the SDKs
+  - These files generally set a number of properties used by the targets
+  - These tend to contain many PropertyGroup and ItemGroup items defining values needed in the builds.
+
+All of these are are [XML Structured](#XML-Structure) text files.
+
+As I noted above, there is nothing truly special with the names and suffixes, they are just standard conventions you should follow.
+
+### Libraries 
+
+Many parts of the build involve invoking outside tools or operations that would be burdensome to write in XML. Task assemblies allow extension of the msbuild by defining types in a C# library that can be called by msbuild targets.
+
+
+- Task Assemblies (.dll)
+  - These files are generally provided by the SDKs
+  - These are C# libraries which reference various msbuild libraries
+  - These define [tasks](#MSBuild-Tasks) which can be invoked inside defined targets to carry out parts of the builds.
+  - They are loaded when other msbuild files use `UsingTask` nodes
+
+
+
+## Data
+
+
+- There is only one data type in msbuild for values. Everything is a string.
+- msbuild distinguished between two types of data, properties and items:
+  - Properties are single named elements 
+    - Think of accessing into a Dictionary<string, string> with the name type being the key
+    - They are defined in PropertyGroups
+
+            <PropertyGroup>
+              <DebugSymbols>true</DebugSymbols>
+            </PropertyGroup>
+
+       - This defines a property "DebugSymbols” with the value “true”
+    - They are referenced later via $(NAME)
```

Comment:
It might be helpful here (and for Items below) to give sample input and output like:
One could refer to the above example with `$(DebugSymbols)`. This would be replaced with `true` before evaluating further. This is particularly helpful in clarifying what the "They may be represented by a semicolon-separated string of items" (and correct that line) part means.